### PR TITLE
fix: correct some typos, following Grammarly (2)

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: lean/main/overview.lean [hunk 1]
```diff
@@ -13,34 +13,34 @@ Metaprogramming in Lean is deeply connected to the compilation steps - parsing,
...
+    Finally, it's time to infuse your syntax with meaning - Lean finds an **elab** that's matched to the appropriate **syntax rule** by the `name` argument (**syntax rules**, **macros** and **elabs** all have this argument, and they must match). The newfound **elab** returns a particular `Expr` object.
     This completes the elaboration step.
 
-Expression (`Expr`) is then converted to the executable code during the evaluation step - we don't have to specify that in any way, Lean compiler will handle that for us.
+The expression (`Expr`) is then converted to the executable code during the evaluation step - we don't have to specify that in any way, the Lean compiler will handle that for us.
 
 ## Elaboration and delaboration
 
-Elaboration is a overloaded term in Lean, for example you can meet the following usage of the word "elaboration", which defines elaboration as *"taking a partially-specified expression and inferring what is left implicit"*:
+Elaboration is an overloaded term in Lean. For example, you can meet the following usage of the word "elaboration", which defines elaboration as *"taking a partially-specified expression and inferring what is left implicit"*:
```
Comment (Julian) [inline lean/main/overview.lean:43]: "meet" is odd phrasing, so something like

```suggestion
Elaboration is an overloaded term in Lean. For example, you might encounter the following usage of the word "elaboration", wherein the intention is *"taking a partially-specified expression and inferring what is left implicit"*:
```

sounds better, but I honestly don't fully understand the sentence -- it's trying to say elaboration has multiple meanings in Lean yeah? And then illustrate one of them? But the word "elaboration" doesn't appear in the quoted text, so where's the other meaning of the word?

**Comment by jcommelin** [comment]
Thanks a lot for those improvements @Julian!

File: lean/main/overview.lean (hunk 1 above)
Comment (Julian) [reply lean/main/overview.lean]: Got it, sounds good!

**Comment by Julian** [comment]
I think this lgtm too after fixing the merge conflict! Thanks for the PRs too.

**Comment by lakesare** [comment]
Oh! Thanks for fixing these @jcommelin and @Julian.

**Pull request merged**